Include/object.h

```diff
@@ -183,9 +231,56 @@ typedef struct {
 PyAPI_FUNC(int) Py_Is(PyObject *x, PyObject *y);
 #define Py_Is(x, y) ((x) == (y))
 
+#ifndef Py_LIMITED_API
```

Comment:
Non-limited API must be in Include/cpython/object.h.

Reply:
That's not possible here and `Include/object.h` already includes a number of non-limited API functions. It's not as cleanly separated as other file pairs. In particular, it's not possible to move this to `Include/cpython/object.h` because the definition is required by `Py_INCREF` and a number of other functions that must be defined before `Include/cpython/object.h` is included.

Reply:
I wrote PR #111505 to fix the inter-dependency issue: move non-limited code to Include/cpython/object.h, and only keep limited code in Include/object.h.